src/osd/PG.cc

```diff
@@ -419,16 +419,9 @@ void PG::queue_recovery()
   } else {
     dout(10) << "queue_recovery -- queuing" << dendl;
     recovery_queued = true;
-    // Let cost per object be the average object size
-    auto num_bytes = static_cast<uint64_t>(
-      std::max<int64_t>(
-	0, // ensure bytes is non-negative
-	info.stats.stats.sum.num_bytes));
-    auto num_objects = static_cast<uint64_t>(
-      std::max<int64_t>(
-	1, // ensure objects is non-negative and non-zero
-	info.stats.stats.sum.num_objects));
-    uint64_t cost_per_object = std::max<uint64_t>(num_bytes / num_objects, 1);
+    uint64_t cost_per_object = std::max<uint64_t>(
```

Comment:
@amathuria Please note that this section is factored out to a method (see PG.h) as part of the snap trim cost PR (see https://github.com/ceph/ceph/pull/54597). This PR should get merged shortly. Therefore, you would need to rebase and call the new method here.

Reply:
@sseshasa got it. I'll wait for that PR to get merged first

Reply:
@sseshasa does this look okay to you now?

Reply:
@amathuria Looks good to me.